src/runtime/vm/vm.cc

```diff
@@ -615,6 +616,26 @@ PackedFunc VirtualMachine::GetFunction(const std::string& name,
     return PackedFunc([sptr_to_self, this](TVMArgs args, TVMRetValue* rv) {
       this->LoadParams(args[0].operator std::string());
     });
+  } else if (name == "get_input") {
+    return PackedFunc([sptr_to_self, this](TVMArgs args, TVMRetValue* rv) {
+        int in_idx = 0;
+        if (args[0].type_code() == kStr) {
+          in_idx = this->GetInputIndex(args[0]);
+        } else {
+          in_idx = args[0];
+        }
+        CHECK_GE(in_idx, 0);
+        *rv = this->GetInput(in_idx);
+      });
+  } else if (name == "run") {
```

Comment:
We can just use `invoke`, right?

Reply:
One caveat is `run` only supports tensor inputs, while `invoke` could be any objects(datatype, tuple). For execution on GPU, I don't think datatype or tuple will be encountered, supporting all kinds of objects complicates copying tensor into device during initialization. So I wrote `run` to handle tensor only function call.